pkg/apis/core/v1beta1/helper/helper.go

```diff
@@ -742,3 +742,19 @@ func WrapWithLastError(err error, lastError *gardencorev1beta1.LastError) error
 	}
 	return errors.Wrapf(err, "last error: %s", lastError.Description)
 }
+
+// IsAPIServerExposureManaged returns true, if the Object is managed by Gardener for API server exposure.
+// This indicates to extensions that they should not mutate the object.
+// Gardener marks the kube-apiserver Service and Deployment as managed by it when it uses SNI to expose them.
+func IsAPIServerExposureManaged(obj metav1.Object) bool {
```

Comment:
Just noticed: Don't you think it'd be more convenient to pass a `runtime.Object` here and use `meta.Accessor(obj)` inside?

Reply:
I don't have to call `meta.Accessor(obj)` to get the object meta

Reply:
Anyway, but what do you thing about passing `runtime.Object` instead of `metav1.Object` so it's more convenient for the caller?

Reply:
You can already do this. Passing some K8S object is already possible:

```go
&gardencorev1alpha1.CloudProfile{
	ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{
		"core.gardener.cloud/apiserver-exposure": "gardener-managed",
	}}
}
```

Reply:
Plus `metav1.Object` is a subset of `runtime.Object`

Reply:
Of course ... thanks for clarifying!